core/chains/evm/client/errors.go

```diff
@@ -333,6 +349,10 @@ func (s *SendError) IsInsufficientEth(configErrors *ClientErrors) bool {
 	return s.is(InsufficientEth, configErrors)
 }
 
+func (s *SendError) IsInvalidSenderError(configErrors *ClientErrors) bool {
```

Comment:
Where is this error being used?

Reply:
@dimriou In `core/chains/evm/client/errors_test.go`

Reply:
I can see that, but where is this actually being used in the code? It seems like we created a new error type just to pass a test case.

Reply:
I created this in preparation for Treasure, so we know we have this case covered

Reply:
Yes, but now we have a new error type that we don't know how it will be handled by the TXM. Let's say we get that error, how is the broadcaster/confirmer going to react to that information? Based on the error message, I get the feeling that it can already be classified under an existing type, probably under fatal. What are the circumstances under which this error pops up and how will the TXM recover from this?

Reply:
I think `InvalidSender` errors can be grouped with `Fatal` like we do with other clients/chains ([example](https://github.com/smartcontractkit/chainlink/blob/c2c31c05ac3fe19d4df8313af25eb740953b935a/core/chains/evm/client/errors.go#L103)). That way we won't have to make any changes to the TXM Broadcaster to pick up on a new error like Dimitris was mentioning above